internal/rollup/rollup_bundle.bzl

```diff
@@ -235,16 +268,81 @@ def run_sourcemapexplorer(ctx, js, map, output):
+  for e in [ctx.attr.entry_point] + ctx.attr.additional_entry_points:
+    entry_point = e.lstrip(build_file_prefix)
+    entry_points["./" + entry_point] = folder + "/" + entry_point.split("/")[-1]
+
+  ctx.actions.expand_template(
+      output = output,
+      template =  ctx.file._system_config_tmpl,
+      substitutions = {
+          "TMPL_entry_points": str(entry_points),
+          "TMPL_main_entry_point": main_entry_point,
+      })
+
 def _rollup_bundle(ctx):
-  rollup_config = write_rollup_config(ctx)
-  run_rollup(ctx, collect_es6_sources(ctx), rollup_config, ctx.outputs.build_es6)
-  _run_tsc(ctx, ctx.outputs.build_es6, ctx.outputs.build_es5)
-  source_map = run_uglify(ctx, ctx.outputs.build_es5, ctx.outputs.build_es5_min)
-  run_uglify(ctx, ctx.outputs.build_es5, ctx.outputs.build_es5_min_debug, debug = True)
-  umd_rollup_config = write_rollup_config(ctx, filename = "_%s_umd.rollup.conf.js", output_format = "umd")
-  run_rollup(ctx, collect_es6_sources(ctx), umd_rollup_config, ctx.outputs.build_umd)
-  run_sourcemapexplorer(ctx, ctx.outputs.build_es5_min, source_map, ctx.outputs.explore_html)
-  files = [ctx.outputs.build_es5_min, source_map]
+  if ctx.attr.additional_entry_points:
+    # Generate code split bundles if additional entry points have been specified.
+    # See doc for additional_entry_points for more information.
+    # Note: ".cs" is needed on the output folders since ctx.label.name + ".es6" is already
+    # a folder that contains the re-rooted es6 sources
+    rollup_config = write_rollup_config(ctx, output_format="cjs", additional_entry_points=ctx.attr.additional_entry_points)
+    code_split_es6_output_dir = ctx.actions.declare_directory(ctx.label.name + ".cs.es6")
+    _run_rollup(ctx, collect_es6_sources(ctx), rollup_config, code_split_es6_output_dir)
+    code_split_es5_output_dir = ctx.actions.declare_directory(ctx.label.name + ".cs")
+    _run_tsc_on_directory(ctx, code_split_es6_output_dir, code_split_es5_output_dir)
+    code_split_es5_min_output_dir = ctx.actions.declare_directory(ctx.label.name + ".cs.min")
+    run_uglify(ctx, code_split_es5_output_dir, code_split_es5_min_output_dir, config_name="%s" % ctx.label.name, is_directory=True)
+    code_split_es5_min_debug_output_dir = ctx.actions.declare_directory(ctx.label.name + ".cs.min_debug")
+    run_uglify(ctx, code_split_es5_output_dir, code_split_es5_min_debug_output_dir, debug=True, config_name="%s-debug" % ctx.label.name, is_directory=True)
+    # Generate the SystemJS boilerplate/entry point files
+    _generate_code_split_entry(ctx, ctx.label.name + ".cs.es6", ctx.outputs.build_es6)
+    _generate_code_split_entry(ctx, ctx.label.name + ".cs", ctx.outputs.build_es5)
+    _generate_code_split_entry(ctx, ctx.label.name + ".cs.min", ctx.outputs.build_es5_min)
+    _generate_code_split_entry(ctx, ctx.label.name + ".cs.min_debug", ctx.outputs.build_es5_min_debug)
+    # There is no UMD bundle when code-splitting but we still need to satisfy the output
```

Comment:
what's the implication of ditching the UMD bundle? I guess you either distribute a library, in which case you need it, or you're an app, and then you don't?
it's sad to have the rule advertise an output that doesn't actually make sense in all usages. Implies maybe there should be a separate rule for rollup_codesplit_bundle (though I also don't like the idea of a bigger API surface area, so ?)

Reply:
If anyone downstream is asking for "bundle.umd.js" as a dependency for another rule that removing that output will break them. I don't know if anyone is using it however. Its not minified. 

Angular language-service has its own `ls_rollup_bundle` which generates a `umd` and `umd.min` although the format is actually `AMD` in that case.

```
# Note: the file is called "umd.js" and "umd.min.js" because of historical
# reasons. The format is actually amd and not umd, but we are afraid to rename
# the file because that would likely break the IDE and other integrations that
# have the path hardcoded in them.
_ROLLUP_OUTPUTS = {
    "build_umd": "%{name}.umd.js",
    "build_umd_min": "%{name}.umd.min.js",
}

def _ls_rollup_bundle(ctx):
    esm5_sources = flatten_esm5(ctx)
    rollup_config = write_rollup_config(
        ctx,
        root_dir = "/".join([ctx.bin_dir.path, ctx.label.package, esm5_root_dir(ctx)]),
        output_format = "amd",
    )
    run_rollup(ctx, esm5_sources, rollup_config, ctx.outputs.build_umd)
    source_map = run_uglify(ctx, ctx.outputs.build_umd, ctx.outputs.build_umd_min)
    return DefaultInfo(files = depset([ctx.outputs.build_umd, ctx.outputs.build_umd_min, source_map]))
```

`ng_package` also has its own rollup rule for generating `umd` and `umd.min`

```
    outputs = {
        "fesm5": "fesm5/%s.js" % basename,
        "fesm2015": "fesm2015/%s.js" % basename,
        "umd": "%s.umd.js" % basename,
        "umd_min": "%s.umd.min.js" % basename,
    }
```

Reply:
Ahhh. The history of that is it was added in https://github.com/bazelbuild/rules_nodejs/pull/170 and was meant to be pushed upstream to ng_package (https://github.com/angular/angular/issues/23134) but never was.